### PR TITLE
Bubble Sort Algorithm 

### PR DESCRIPTION
**Pull request by Tahirbhalli**
This is the bubble sort algorithm created in Ruby.

### CODE REVIEW

File: buble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                temp=arr[i]
+                arr[i]=arr[i-1]
+                arr[i-1]=temp
+            end
+            i=i+1
+        end
+        a=a+1
+    end
+  
+end
```
Comment (shshamim63) [inline buble_sort.rb]: The naming of the method is incorrect. Cause you were asked to implement bubble_sort method ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sort.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1)). But instead I can see the name as `bublesort`. So please rename the function. The main task of the bubble_sort is to return a modified array. But I can not see any arr at the line no. 19. So please add `arr` at line no. 19.

File: buble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            result=yield(r[j],r[j+1])
+            next unless result.positive?
+            temp=r[j]
+            r[j]=r[j+1]
+            r[j+1]=temp
+        end
+    end
+    for element in r
+        print (element.to_s+" ")
+    end
```
Comment (shshamim63) [inline buble_sort.rb]: The task of the bubble_sort _by is to return the modified array. If you look at the default sorting then you will see that it returns an array. But on your one it prints the element of the array. Also the naming on the variable is not self-descriptive . What do you mean by `r`. You can name that as you did on the bubble_sort like `arr`.

**Review by shshamim63 — COMMENTED** [review]
## Status :space_invader: Required Changes :space_invader: 
___
Hi @Tahirbhalli and @sddoe :wave: 
Awesome initiative :clap: . You can see you have handled the loop differently in both of the methods. That is a tricky job you have covered. But there are some issues that are actually blocking me to approve this project right now. And I am sure that you will fix them easily. So lets look at the issues.
___
### :spades: [Mandatory Changes] :spades: 
___
1. I can not see any description on readme in this project. I deem you have forgot to add it. You can add a sort description with a pseudo-code. Along with that you can add how to contact with the contributor, Do you accept PR? And so on.
2. Well the .rubocop.yml file seems different from the one provided from microverse. You are not suppose to change the rubocop file. Check this [link](https://github.com/microverseinc/linters-config/tree/master/ruby) to reset your .rubocop.yml. Cause I can see many offenses yelling by rubocop. 
3. Bubble_sort method is not working like it should. Check out the comment below.
4. Bubble_sort_by also not working like it should. Please check the comment below.  
___
#### Please resolve the issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)

File: buble_sort.rb (hunk 1 above)
Comment (Tahirbhalli) [reply buble_sort.rb]: Done

File: buble_sort.rb (hunk 2 above)
Comment (Tahirbhalli) [reply buble_sort.rb]: Done

**Review by bolah2009 — APPROVED** [review]
It's a 👍🏾 from me!

### :heavy_check_mark: Approved!

Nice job implementing the changes suggested. :relaxed: 

Good luck with your next projects. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Pull request merged**